testonly/hammer/record.go

```diff
@@ -83,13 +84,11 @@ func (g *sharedState) advertiseSMR(smr types.MapRootV1) error {
 		if g.smrs[0] == nil || g.smrs[0].Revision < smr.Revision {
 			return true, nil
 		}
-
-		for i := 0; i < smrCount && g.smrs[i] != nil && g.smrs[i].Revision >= smr.Revision; i++ {
-			if g.smrs[0].Revision == smr.Revision {
-				if !reflect.DeepEqual(g.smrs[i], &smr) {
-					return false, fmt.Errorf("SMR mismatch at revision %d: had %+v, got %+v", smr.Revision, g.smrs[0], smr)
-				}
-			}
+		pos := sort.Search(smrCount, func(i int) bool {
```

Comment:
Note that there is a "not found" `pos == smrCount` situation, if all `smrs` have `Revision > smr.Revision`. That's why I put "minus 1" in my suggestion, which helped to mitigate this subtlety. It's a bit ninja though, so feel free to comment it /  make it readable in other way.

Reply:
Alternatively, you could allocate an extra "sentinel" item in the `smrs` slice.